pkg/kubelet/cm/cpumanager/policy_static.go

```diff
@@ -486,7 +486,7 @@ func (p *staticPolicy) takeByTopology(availableCPUs cpuset.CPUSet, numCPUs int)
 	if p.options.DistributeCPUsAcrossNUMA {
 		cpuGroupSize := 1
 		if p.options.FullPhysicalCPUsOnly {
-			cpuGroupSize = p.topology.CPUsPerCore()
+			cpuGroupSize = p.topology.MaxCPUsPerCore()
 		}
```

Comment:
Is this going to work? I'd need to refresh my memory on how the algorithm works, but f we encounter a core that has less than `MaxCPUsPerCore()` isn't that going to cause problems? I'm pretty sure we allocate CPUs from a core as a group of size `cpuGroupSize`.  So if one of the cores has less than `MaxCPUsPerCore()`, the accounting is going to be off.

Reply:
It seems to me that a good (only?) way forward is to have a way to preserve the cpu-sibling information and use it in `takeByTopologyNUMADistributed` when computing the best combination

Reply:
yes, if we define core0 with 1CPU offline as a free core, we need to refactor these logic. but i think it's rather compliated since it is designed to work with cpugroupsize parameter other than real topo.

on the other hand if we define core0 as a non-free core, `takeByTopologyNUMADistributed` can stay the same. To achieve that, `IsCoreFree` needs to be modified from comparing free CPU nums with `a.topo.CPUDetails.CPUsInCores(coreID).Size()` to comparing free CPU nums with something like `MaxCPUsPerCore`

Reply:
I'm not pushing for invasive changes in this PR, so I'd be very in favor of a solution along the lines of what we have now. I'm just concerned such a solution may not feasible with the current architecture and with the available topo data. Let's see.

Reply:
Got it. Any further opnion when have time @klueska